Cognite.Simulator.Utils/ModelStateBase.cs

```diff
@@ -47,6 +47,17 @@ public virtual bool ShouldProcess()
             return !string.IsNullOrEmpty(FilePath) && CanRead && !isParsedBefore;
         }
 
+        /// <summary>
+        /// Indicates if the model file has been downloaded and the file exists on the disk.
+        /// </summary>
+        public bool Downloaded
```

Comment:
To me, this implementation is a little scary. I don't know exactly where this value is checked, but what happens if a file is in the _process_ of being downloaded? Won't this return `true` in that case? Potentially incorrectly informing consumers that the file is ready?

Reply:
nope. we only set filepath after something has been downloaded, not before / in the process